### PR TITLE
feat: support configuring nodeSelectors and tolerations for tunnel resources 

### PR DESCRIPTION
**Pull request by shebpamm**
This PR introduces two changes to the tunnel.spec:
`nodeSelectors` & `tolerations`. 

These get passed to the cloudflared deployment. 

### Things of imporance:
- The `tolerations` field is directly of type `[]corev1.Toleration` which shows as []Object in the tunnel.spec. This could be improved.
- By default these values are omitted
- Tested on a testing cluster

### Notes on the PR:
- No docs changed yet, will add once the code is reviewed
- autogenerated kubebuilder files in a separate commit for easier review

@adyanth

### CODE REVIEW

File: controllers/reconciler.go [hunk 1]
```diff
@@ -405,6 +409,9 @@ func secretForTunnel(r GenericTunnelReconciler) *corev1.Secret {
 func deploymentForTunnel(r GenericTunnelReconciler) *appsv1.Deployment {
 	ls := labelsForTunnel(r.GetTunnel())
 	replicas := r.GetTunnel().GetSpec().Size
+	nodeSelector := nodeSelectorsForTunnel(r.GetTunnel())
+	tolerations := r.GetTunnel().GetSpec().Tolerations
+	r.GetLog().Info("NodeSelectors", "NodeSelectors", nodeSelector)
```
Comment (adyanth) [inline controllers/reconciler.go:414]: The config need not be logged back

File: config/crd/bases/networking.cfargotunnel.com_clustertunnels.yaml [hunk 2]
```diff
@@ -159,9 +204,3 @@ spec:
     storage: true
     subresources:
       status: {}
-status:
-  acceptedNames:
-    kind: ""
-    plural: ""
-  conditions: []
-  storedVersions: []
```
Comment (adyanth) [inline config/crd/bases/networking.cfargotunnel.com_clustertunnels.yaml:167]: Do you know why these were removed with your changes/kubebuilder version?

[The kubebuilder book](https://book-v1.book.kubebuilder.io/basics/status_subresource) seems to say it should be present when marked with `//+kubebuilder:subresource:status`

**Review by adyanth — CHANGES_REQUESTED** [review]
Thank you for contributing! 

This looks good (other than minor comments). The tunnel should ideally take in a `PodSpec` override, but I have not had the use case for it yet. NodeSpec and Tolerations are the most common ones, so thanks for exposing it.

File: controllers/reconciler.go (hunk 1 above)
Comment (shebpamm) [reply controllers/reconciler.go]: Indeed, change snuck in from testing. Addressed in [2cfbf67](https://github.com/adyanth/cloudflare-operator/pull/98/commits/2cfbf6712d8770deec8ea079b83b3597a19d1f99)

File: config/crd/bases/networking.cfargotunnel.com_clustertunnels.yaml (hunk 2 above)
Comment (shebpamm) [reply config/crd/bases/networking.cfargotunnel.com_clustertunnels.yaml:167]: I'm able to reproduce this lack of status with the proper (0.7.0) version of controller-tools as well. Could you try reproducing it?

Comment (shebpamm) [reply config/crd/bases/networking.cfargotunnel.com_clustertunnels.yaml:167]: After even more testing, it seems like the version of kubebuilder is always v0.13.0 even if I use an older controller-gen version, seems to be my own setup. I'm a bit lost on how to solve this though.

Comment (adyanth) [reply config/crd/bases/networking.cfargotunnel.com_clustertunnels.yaml:167]: Will try and get back to you asap

Comment (adyanth) [reply config/crd/bases/networking.cfargotunnel.com_clustertunnels.yaml:167]: With controller-gen v0.7.0, it shows the status section, but v0.15.0 does not. I even created a new project as per the kubebuilder book and it does not match what they show. 

For now, could you use the v0.7.0 and run make manifests? I will try to update the whole setup to use the latest version of everything soon (TM) https://github.com/adyanth/cloudflare-operator/issues/99

Comment (shebpamm) [reply config/crd/bases/networking.cfargotunnel.com_clustertunnels.yaml]: Now built with controller-gen v0.7.0! Addressed in [ca50b9d](https://github.com/adyanth/cloudflare-operator/pull/98/commits/ca50b9d61fe615aaf84099b880be5c530d3f9e4d)

**Review by adyanth — APPROVED** [review]

**Pull request merged**